cores/esp32/USBCDC.cpp

```diff
@@ -246,14 +251,27 @@ void USBCDC::_onLineCoding(uint32_t _bit_rate, uint8_t _stop_bits, uint8_t _pari
 void USBCDC::_onRX(){
     uint8_t buf[CONFIG_TINYUSB_CDC_RX_BUFSIZE+1];
     uint32_t count = tud_cdc_n_read(itf, buf, CONFIG_TINYUSB_CDC_RX_BUFSIZE);
+
+    if(rx_queue == NULL) {
+        return;
+    }
+    if (uxQueueSpacesAvailable(rx_queue) < count) {
+        //this VTaskDelay gives, to Arduino's task, time to the CPU do its processing
+        //without it, data may be lost when the number of bytes received is higher than CDC buffer size
+        vTaskDelay(10);
+    }
     for(uint32_t i=0; i<count; i++){
-        if(rx_queue == NULL || !xQueueSend(rx_queue, buf+i, 0)){
-            return;
+        if(!xQueueSend(rx_queue, buf+i, 0)){
+            // rx_queue overflow - data will be lost
+            count = i;
+            break;
         }
     }
-    arduino_usb_cdc_event_data_t p;
-    p.rx.len = count;
-    arduino_usb_event_post(ARDUINO_USB_CDC_EVENTS, ARDUINO_USB_CDC_RX_EVENT, &p, sizeof(arduino_usb_cdc_event_data_t), portMAX_DELAY);
+    if (count) {
```

Comment:
whe would it be possible count to be 0?

Reply:
When no data is read by Arduino Task and the Queue continues full after ```vTaskDelay();```
In that case the ```for()``` will breakin the first ```if()``` test and count will be 0.